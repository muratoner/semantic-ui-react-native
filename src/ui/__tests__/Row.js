import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import React from 'react'
import {StyleSheet} from 'react-native'
import {Avatar, Row} from '../../'

describe('Row Component', () => {
	it('should render without issues', () => {
		const component = shallow(<Row style={{backgroundColor: 'red'}} />)

		expect(component.length).toBe(1)
		const style = StyleSheet.flatten(component.get(0).props.style)
		expect(style).toHaveProperty('flexDirection', 'row')
		expect(style).toHaveProperty('backgroundColor', 'red')
		expect(toJson(component)).toMatchSnapshot()
	})

	it('should render with children', () => {
		const component = shallow(
			<Row>
				<Avatar title='Murat Ã–NER' />
				<Avatar title='Murat' />
			</Row>
		)

		expect(component.length).toBe(1)
		expect(component.get(0).props.children.length).toBe(2)
		expect(toJson(component)).toMatchSnapshot()
	})
})
