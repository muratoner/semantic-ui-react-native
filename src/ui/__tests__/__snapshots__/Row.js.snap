// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Component should render with children 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <Component
    title="Murat Ã–NER"
  />
  <Component
    title="Murat"
  />
</View>
`;

exports[`Row Component should render without issues 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
/>
`;
